call-profile/main.go

```diff
@@ -0,0 +1,566 @@
+		validation.Field(&res.Email, validation.Required, is.Email),
+		validation.Field(&res.YOE, validation.Min(0)),
+		validation.Field(&res.Company, validation.Required),
+		validation.Field(&res.Designation, validation.Required),
+		validation.Field(&res.GithubId, validation.Required),
+		validation.Field(&res.LinkedIn, validation.Required),
+		validation.Field(&res.Website, is.URL))
+}
+
+/*
+ MODELS
+*/
+
+/*
+ Logs the health of the user's service
+*/
+func logHealth(client *firestore.Client, ctx context.Context, userId string, isServiceRunning bool, sessionId string) {
+	newLog := Log{
+		Type:      Constants["PROFILE_SERVICE_HEALTH"],
+		Timestamp: time.Now(),
+		Meta: map[string]interface{}{
+			"userId":    userId,
+			"sessionId": sessionId,
+		},
+		Body: map[string]interface{}{
+			"userId":         userId,
+			"serviceRunning": isServiceRunning,
+		},
+	}
+	client.Collection("logs").Add(ctx, newLog)
+}
+
+/*
+ Logs the status of the user's profileDiff
+*/
+func logProfileSkipped(client *firestore.Client, ctx context.Context, userId string, reason string, sessionId string) {
+	newLog := Log{
+		Type:      Constants["PROFILE_SKIPPED"],
+		Timestamp: time.Now(),
+		Meta: map[string]interface{}{
+			"userId":    userId,
+			"sessionId": sessionId,
+		},
+		Body: map[string]interface{}{
+			"userId": userId,
+			"reason": reason,
+		},
+	}
+	client.Collection("logs").Add(ctx, newLog)
+}
+
+func logProfileStored(client *firestore.Client, ctx context.Context, userId string, sessionId string) {
+	newLog := Log{
+		Type:      Constants["PROFILE_DIFF_STORED"],
+		Timestamp: time.Now(),
+		Meta: map[string]interface{}{
+			"userId": userId,
+			"sessionId": sessionId
+		},
+		Body: map[string]interface{}{
+			"userId": userId,
+		},
+	}
+	client.Collection("logs").Add(ctx, newLog)
+}
+
+/*
+ Function for setting the profileStatus in user object in firestore
+*/
+func setProfileStatusBlocked(client *firestore.Client, ctx context.Context, userId string, reason string, sessionId string) {
+	client.Collection("users").Doc(userId).Set(ctx, map[string]interface{}{
+		"profileStatus": Constants["STATUS_BLOCKED"],
+		"chaincode":     "",
+	}, firestore.MergeAll)
+
+	newLog := Log{
+		Type:      Constants["PROFILE_SERVICE_BLOCKED"],
+		Timestamp: time.Now(),
+		Meta: map[string]interface{}{
+			"userId":    userId,
+			"sessionId": sessionId,
+		},
+		Body: map[string]interface{}{
+			"userId": userId,
+			"reason": reason,
+		},
+	}
+	client.Collection("logs").Add(ctx, newLog)
+}
+
+/*
+ sets the user's profile diff to not approved
+*/
+func setNotApproved(client *firestore.Client, ctx context.Context, lastdiffId string) {
+	client.Collection("profileDiffs").Doc(lastdiffId).Set(ctx, map[string]interface{}{
+		"approval": Constants["NOT_APPROVED"],
+	}, firestore.MergeAll)
+}
+
+/*
+ Get the last profile diff of the user
+*/
+func getLastDiff(client *firestore.Client, ctx context.Context, userId string, approval string) (Res, string) {
```

Comment:
Ref.ID field of the Document object returns the ID of the document. However, the lastdiffId variable is declared as a string. This means that the value of the Ref.ID field cannot be directly assigned to the lastdiffId variable.

Reply:
The type of ID of the document object is String. I tested it.